src/typeChecker/utils.ts

```diff
@@ -352,16 +362,41 @@ export const temporaryStreamFuncs: [string, BindableType][] = [
 
 // Prelude function type overrides for Source Typed variant
 export const source1TypeOverrides: [string, BindableType][] = [
-  // is something functions
-  ['is_boolean', tFunc(tAny, tBool)],
-  ['is_number', tFunc(tAny, tBool)],
-  ['is_string', tFunc(tAny, tBool)],
-  ['is_undefined', tFunc(tAny, tBool)],
-  ['is_function', tFunc(tAny, tBool)],
+  // math functions
+  // TODO: Add support for type checking of functions with variable no. of args
+  ['math_hypot', tForAll(tNumber)],
+  ['math_max', tForAll(tNumber)],
+  ['math_min', tForAll(tNumber)],
   // misc functions
   ['stringify', tFunc(tAny, tString)],
   ['arity', tFunc(tAny, tNumber)],
-  ['char_at', tFunc(tString, tNumber, tUnion(tString, tUndef))]
+  ['char_at', tFunc(tString, tNumber, tUnion(tString, tUndef))],
+  // TODO: Add support for type checking of functions with variable no. of args
+  ['display', tForAll(tAny)],
+  ['error', tForAll(tAny)]
+]
+
+export const source2TypeOverrides: [string, BindableType][] = [
+  // list library functions
+  ['accumulate', tFunc(tFunc(tAny, tAny, tAny), tAny, tList(tAny), tAny)],
+  ['append', tFunc(tList(tAny), tList(tAny), tList(tAny))],
+  ['build_list', tFunc(tFunc(tAny, tAny), tNumber, tList(tAny))],
+  ['enum_list', tFunc(tNumber, tNumber, tList(tNumber))],
+  ['filter', tFunc(tFunc(tAny, tAny), tList(tAny), tList(tAny))],
+  ['for_each', tFunc(tFunc(tAny, tAny), tList(tAny), tBool)],
+  ['length', tFunc(tList(tAny), tNumber)],
+  ['list_ref', tFunc(tList(tAny), tNumber, tAny)],
+  ['list_to_string', tFunc(tList(tAny), tString)],
+  ['map', tFunc(tFunc(tAny, tAny), tList(tAny), tList(tAny))],
+  ['member', tFunc(tAny, tList(tAny), tList(tAny))],
+  ['remove', tFunc(tAny, tList(tAny), tList(tAny))],
+  ['remove_all', tFunc(tAny, tList(tAny), tList(tAny))],
+  ['reverse', tFunc(tList(tAny), tList(tAny))],
+  // misc functions
+  // TODO: Add support for type checking of functions with variable no. of args
+  ['display_list', tForAll(tAny)],
+  ['draw_data', tForAll(tAny)],
+  ['equal', tFunc(tAny, tAny, tBool)]
 ]
```

Comment:
To confirm, this means that once I run any kind of operation on lists, I lose the type of its elements?

Reply:
Unfortunately as of right now the type error checker is not yet powerful enough to infer the return type based on the types of its arguments. I will KIV this for future enhancements.